src/dmd/constfold.d

```diff
@@ -120,6 +120,9 @@ UnionExp Not(Type type, Expression e1)
 {
     UnionExp ue = void;
     Loc loc = e1.loc;
+    // BUG: Should be replaced with e1.toBool().get(), but this is apparently
+    //      executed for some expressions that cannot be const-folded
+    //      To be fixed in another PR
     emplaceExp!(IntegerExp)(&ue, loc, e1.toBool().hasValue(false) ? 1 : 0, type);
```

Comment:
Using `get` here triggers an assertion failure for this code:
```d
bool bug4065()
{
    int[string] aa = ["aa":14];
    int* p = "aa" in aa;
    return !p;
}

static assert(!bug4065());
```

The problem here is that `e1` is an `AddrExp` that - unlike `SymOffExp` - doesn't override `toBool`. `SymOffExp` assumes that it's always `true`, so maybe `AddrExp` should do the same?